hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -350,6 +368,63 @@ public void createDbStore(MutableVolumeSet dbVolumeSet) throws IOException {
     }
   }
 
+  /**
+   * Ensure that volume is initialized properly with
+   * cleanup path. Should disk be re-inserted into
+   * cluster, cleanup path should already be on
+   * disk. This method syncs the HddsVolume
+   * with the path on disk.
+   * 
+   * @param id clusterId or scmId
+   */
+  public void checkTmpDirPaths(String id) {
+    Path tmpPath = createTmpPath(id);
+    deleteServiceDirPath = tmpPath.resolve(TMP_DELETE_SERVICE_DIR);
+  }
+
+  private Path createTmpPath(String id) {
+
+    // HddsVolume root directory path
+    String hddsRoot = getHddsRootDir().toString();
+
+    // HddsVolume path
+    String vol = HddsVolumeUtil.getHddsRoot(hddsRoot);
+
+    Path volPath = Paths.get(vol);
+    Path idPath = Paths.get(id);
+
+    return volPath.resolve(idPath).resolve(TMP_DIR);
+  }
+
+  private void createTmpDir(String id) {
```

Comment:
update interface, unnecessary `id` argument. (clusterid defined in HddsVolume / StorageVolume)